### PR TITLE
fix: Split Expense Service

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 739411c</samp>

The pull request enhances the type safety and consistency of the expense model and the split expense service by using more specific interface types instead of `any`. It also adds two new properties to the `FileObject` interface to store the file purpose and content.

### <samp>🤖 Generated by Copilot at 739411c</samp>

> _We are the masters of the code, we shape it as we please_
> _We use specific types to crush the bugs and errors_
> _We move and change the interfaces, we add new properties_
> _We are the SplitExpenseService, we split the bills of terrors_

### <samp>🤖 Generated by Copilot at 739411c</samp>

*  Enforce consistent types for transaction currency, amount, and original currency and amount in `Expense` interface ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L55-R55), [link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L107-R108))
* Move `FileObject` interface from `file-obj.model.ts` to `expense.model.ts` and add `purpose` and `content` properties ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-02c1b013a9189a6b73a85a5032757470ccf95b950701490c6618a45af21b63f5L10-R11))
* Use `FileObject` interface type for file objects in `SplitExpenseService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L61-R61), [link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L72-R73))
* Use `PolicyViolationComment` interface type for policy violation comments in `SplitExpenseService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L124-R124))
* Use `PublicPolicyExpense` interface type for policy expenses in `SplitExpenseService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L214-R214))
* Use `Transaction` interface type for transactions and split expenses in `SplitExpenseService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L231-R231), [link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L258-R259), [link](https://github.com/fylein/fyle-mobile-app/pull/2105/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L311-R315))

## Clickup
https://app.clickup.com/t/85zt699qx

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Missing method return types

File: src/app/core/services/split-expense.service.ts [hunk 1]
```diff
@@ -69,8 +69,8 @@ export class SplitExpenseService {
     });
 
     return forkJoin(fileObservables).pipe(
-      map((data: any[]) => {
-        newFileObjs.forEach((fileObj, index) => {
+      map((data) => {
```
Comment (Dimple16) [inline src/app/core/services/split-expense.service.ts:72]: is removing type here intentional?

**Review by Dimple16 — APPROVED** [review]
one question, rest lgtm

**Review by yksolanki9 — APPROVED** [review]

File: src/app/core/services/split-expense.service.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/core/services/split-expense.service.ts:72]: Yes removing any

**Pull request merged**